### PR TITLE
Bump NodeJS in base image to 18

### LINKED ISSUE #148: Inconsistent Node versions between images, not using latest LTS
### What happened?

I'm using the docker images to run applications using nodejs.
The `pulumi/pulumi` image uses node `v16.20.1`, while the `pulumi/pulumi-nodejs` uses `v18.16.0`.

### Expected Behavior

I was expecting to see the same node version across the different images. Moreover,  both the images are not using the current node LTS version.

### Steps to reproduce

```bash
docker run -it --entrypoint node public.ecr.aws/pulumi/pulumi-nodejs --version
docker run -it --entrypoint node public.ecr.aws/pulumi/pulumi --version
```

### Output of `pulumi about`

```
CLI
Version      3.73.0
Go Version   go1.20.5
Go Compiler  gc

Host
OS       ubuntu
Version  22.04
Arch     x86_64
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by dixler**
This PR upgrades Node.js in the `pulumi/pulumi` image and `pulumi/nodejs` UBI image to the Active LTS version 18, and documents our version policy to keep language runtimes up-to-date with LTS versions.

Fixes #148

### CODE REVIEW

**Review by Frassle — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Images are pushed to:
 - .NET 6.0
 - Go 1.20
 - JDK 11
-- Node.js 16
+- Node.js 18
```
Comment (dixler) [inline README.md:40]: @justinvp Should I put
```suggestion
- Node.js LTS (currently v18)
```

or do we want to put something more extensive?

File: docker/pulumi/Dockerfile [hunk 2]
```diff
@@ -39,7 +39,7 @@ RUN apt-get update -y && \
   rm -rf aws && \
   rm awscliv2.zip && \
   # Add additional apt repos all at once
-  echo "deb https://deb.nodesource.com/node_16.x $(lsb_release -cs) main"                         | tee /etc/apt/sources.list.d/node.list             && \
+  echo "deb https://deb.nodesource.com/node_18.x $(lsb_release -cs) main"                         | tee /etc/apt/sources.list.d/node.list             && \
```
Comment (komalali) [inline docker/pulumi/Dockerfile:42]: how do we make sure we don't get caught behind again? looks like Node 20 is [becoming LTS starting October 24th](https://github.com/nodejs/release#release-schedule:~:text=2023%2D04%2D18-,2023%2D10%2D24,-2024%2D10%2D22)

File: README.md [hunk 3]
```diff
@@ -37,9 +37,13 @@ Images are pushed to:
...
 - Python 3.9
 
+### NodeJS LTS
+
+Our Docker images aim to use NodeJS LTS version (v18 as of writing). You can pin the image tag to a particular version in order to avoid unintended upgrades.
+
```
Comment (justinvp) [inline README.md:46]: Let's document the overall policy for all language runtimes, not just Node. I'll add to the commit.

**Review by justinvp — APPROVED** [review]
LGTM

**Comment by dixler** [comment]
Commits look good.

File: docker/pulumi/Dockerfile (hunk 2 above)
Comment (justinvp) [reply docker/pulumi/Dockerfile:42]: Opened https://github.com/pulumi/pulumi-docker-containers/issues/158 to track.

**Pull request merged**